### PR TITLE
refactor: decrease the cli request timeout

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
while the server isn't reachable, cli takes a long time to timeout  

**Solution:**
decrease the timeout for server reachable check

**Related Issue:**
#1960

### CODE REVIEW

**Review by shanewxy — DISMISSED** [review]
LGTM

File: pkg/cli/config/config.go [hunk 1]
```diff
@@ -196,9 +202,9 @@ func (c *Config) ServerVersion() (*Version, error) {
 		return nil, err
 	}
 
-	resp, err := c.DoRequest(req)
+	resp, err := c.DoRequestWithTimeout(req, fastTimeout)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("access %s failed", c.Server)
```
Comment (gitlawr) [inline pkg/cli/config/config.go:207]: Do we need to print original error

File: pkg/cli/config/config.go [hunk 2]
```diff
@@ -18,7 +18,8 @@ import (
 )
 
 const (
-	timeout     = 15 * time.Second
+	fastTimeout = 1 * time.Second
```
Comment (thxCode) [inline pkg/cli/config/config.go:21]: what means `fastTimeout`, can we use shortTimeout?

File: pkg/cli/config/config.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/cli/config/config.go:207]: No need, while use --debug flag, will log the response, and can check the failed reason by response body.

File: pkg/cli/config/config.go [hunk 3]
```diff
@@ -18,10 +18,11 @@ import (
 )
 
 const (
-	timeout     = 15 * time.Second
-	openAPIPath = "/openapi"
-	apiVersion  = "v1"
-	versionPath = "/debug/version"
+	shortTimeout = 1 * time.Second
+	timeout      = 5 * time.Second
```
Comment (thxCode) [inline pkg/cli/config/config.go:22]: @aiwantaozi  the `client.Timeout` controls the request from dial to response header reading, I think 1s is too short for some bad networking connection cases.

we should configure more granular timeouts, here is an example.
```
&http.Client{
    Transport: &http.Transport{
        Proxy: http.ProxyFromEnvironment,
        DialContext: (&net.Dialer{
            Timeout:   1 * time.Second,           // shorten this to make sure if we can connect to server
            KeepAlive: 30 * time.Second,  
            DualStack: true,
        }).DialContext,
        TLSHandshakeTimeout:   10 * time.Second,  // server must finish cert generation if using self-signed CA
        ResponseHeaderTimeout: 30 * time.Second,  // server must complete the request and start to transit the response 
    }
}
```

Comment (aiwantaozi) [reply pkg/cli/config/config.go]: updated

**Review by thxCode — APPROVED** [review]
LGTM
